contracts/libs/0.8.x/BytecodeStorage.sol

```diff
@@ -175,6 +177,40 @@ library BytecodeStorage {
         }
     }
 
+    /**
+     * @notice Get address for deployer for given contract bytecode
+     * @param _address address of deployed contract with bytecode containing concat(gated-cleanup-logic, data)
+     * @return writerAddress address read from contract bytecode
+     */
+    function getWriterAddressForBytecode(address _address)
+        internal
+        view
+        returns (address writerAddress)
+    {
+        // get the size of the data
+        uint256 codeSize = _codeSizeAt(_address);
+        // handle case where address contains code < DATA_OFFSET
+        // note: the first check here also captures the case where codeSize == 0
+        //       implicitly, but we add the second check of (codeSize == 0)
+        //       as a fall-through that will never execute unless `DATA_OFFSET`
+        //       is set to 0 at some point.
+        if ((codeSize < DATA_OFFSET) || (codeSize == 0)) {
+            revert("ContractAsStorage: Read Error");
+        }
+
+        assembly {
+            // allocate free memory
+            writerAddress := mload(0x40)
```

Comment:
writerAddress should already be allocated since it is in the return statement. If we did allocate here, we would also need to increment the free memory pointer by 0x20, but I think we can skip altogether.